src/private/Write-AzOpsLog.ps1

```diff
@@ -9,25 +13,31 @@ function Write-AzOpsLog {
         [Parameter(Mandatory = $false)]
         [string]$Topic,
 
-        [Parameter(Mandatory = $true)]
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
         [string]$Message,
 
         [Parameter(Mandatory = $false)]
-        [string]$Timestamp
+        [switch]$Timestamp
     )
 
     begin {
-        if ($Timestamp) {
-            $timestamp = Get-Date -Format "[MM/dd/yyyy - HH:mm]"
-            $log = ($timestamp + " " + "($topic)" + " " + $message)
+        # Generate log message prefix with Timestamp (optional) and Topic (optional)
+        # Will apply the same values to all messages sent via pipeline
+        # Uses AzOpsLogTimestampPreference variable to set defaults
+        $messagePrefix = ""
+        if ($Timestamp -or $Global:AzOpsLogTimestampPreference) {
+            $logTimeUtc = Get-Date -Format FileDateTimeUniversal
```

Comment:
Can we explore options for making this output little more readable?
"20200619T0907562622Z"

Reply:
Updated code to: `(Get-Date).ToUniversalTime().ToString("yyyy-MM-dd HH:mm:ss.ffff")`

To provide format like: `2020-06-19 09:33:57.2717` in message.